validation/README.md

```diff
@@ -0,0 +1,112 @@
+# Validation
+
+This document is notes on what has been done so far, and is not intended for `main` in its current form.
+
+## Input
+
+This is designed to take variant input in the form of a MatrixTable, as it is coupled with the core CPG pipeline which has MT as its main output data.
+
+In addition to the variant data, it also sources a number of items from Metamist:
+
+- Path to a 'Truth' VCF
+- Path to a corresponding BED of confident regions
+- Mapping of External:CPG IDs for the cohort
+
+### Variant Data
+
+A MT is provided as input, but a VCF is required for the tools to operate on. The first step is to pull the validation samples from Metamist, pull the samples from the MT, and for all common samples create a single-sample VCF. This is done with the following considerations (per sample):
+
+- subset the MT columns to the single sample, then remove all sites where the remaining sample is HomRef
+- write the remaining sites as a VCF.bgz to a `single_sample_vcf` directory, with the file containing the sample ID. These files will be persisted, and the locations recorded in Metamist for the analysis run
+- `Undecided`: should all sites not indicated as PASS be removed? These are ignored by the validation software by default, so currently are being kept for debugging results.
+
+### Metamist
+
+Metamist is the database for the validation operations, linking each sample with the relevant data.
+
+1. Sample IDs
+    - Within metamist, samples are identified by both the external and internal IDs. Within the MT dump, the samples are referred to by their External IDs, whilst query operations on metamist objects are based on the CPG IDs. The first Metamist query is to pull a mapping of the Internal:External IDs through the `get_external_participant_id_to_internal_sample_id` method. Using this we can swap between IDs interchangeably.
```

Comment:
When we want to include external IDs, I like the idea of using "combined ids" where CPG ID and external ID are merged together, so you don't have to refer to metamist when manually exploring data
<img width="428" alt="Screen Shot 2022-08-19 at 11 56 42 am" src="https://user-images.githubusercontent.com/1575412/185525842-96c5a01d-84a5-4ed0-b5f4-d6518bc333c5.png">

I use this function for that:
https://main-web.populationgenomics.org.au/validation/qc/cram/multiqc.html
https://github.com/populationgenomics/production-pipelines/blob/initial/cpg_pipes/targets.py#L106-L110